src/solvers/simplex/diophantine_systems.c

```diff
@@ -350,6 +350,7 @@ static dcolumn_t *column_add_unit(dcolumn_t *c, int32_t i, int32_t k) {
 static void column_reduce_factor(rational_t *a, dcolumn_t *c1, dcolumn_t *c2) {
   assert(active_row(c1) == active_row(c2));
   q_set(a, active_coeff(c1));
+  q_normalize(active_coeff(c2));
```

Comment:
this seems unrelated to ff. what does this change fix?

Reply:
See change in `q_integer_div` and `q_integer_mod`: originally, none of the parameter of `q_integer_div` was const, even though one would expect the parameter `r2` to be constant. This was because `r2` was normalized inside these functions. However, I needed the 2nd parameter to be constant to ensure the finite field order is not changed inadvertently.
Thus, I removed this normalizing step from `q_integer_div` and `q_integer_mod`, added a comment to these functions, and, in order to preserve original behavior, added the normalizing step before any call to the modified functions in existing code.